ogr/ogrsf_frmts/sqlite/ogrsqlitetablelayer.cpp

```diff
@@ -1710,6 +1717,13 @@ OGRSQLiteTableLayer::CreateGeomField(const OGRGeomFieldDefn *poGeomFieldIn,
         }
     }
 
+    if (m_poDS->IsInTransaction())
```

Comment:
There's a bit of dissymetry with the call to AddFieldDefn() in CreateField() having been done *before* the "if IsInTransation())" logic and here where AddGeomFieldDefn() is done *after* the IsInTransaction() test  (note that if you symmetrize things you need to add/remove a -1 index to GetFieldCount/GetGeomFieldCount)

Reply:
the reason for the dissimetry is because poGeomField is (already was) a unique ptr whic would be null after the move. I will add a comment.

Reply:
comment added